spec/VehicleSignalSpecification.vspec

```diff
@@ -166,6 +166,17 @@ Vehicle.Driver:
 
 #include Driver/Driver.vspec Vehicle.Driver
 
+#
+# Occupant Data
+#
+
+Vehicle.Occupant:
```

Comment:
I find this makes a lot of assumptions about occupant position. I am also concerned that it conflates different concepts (occupant and their position) together. That is, it forces a 1:1 match between occupant and their seat position.

I would have expected occupants to be a dynamic list of persons that is independent (instance-wise) from their position. With this approach, each occupant would reference a seat, while seats would reference occupants. Something like:

```
Vehicle.Occupant:
  type: branch
  instances:
    - Person[1,10]

Vehicle.Occupant.SeatPosition:
  type: reference
  description: A reference to the Occupant's seat position as a node path name (e.g. Vehicle.Cabin.Seat.Row1.Pos1).

Vehicle.OccupantCount:
  datatype: uint8
  type: sensor
  description: The current number of persons in the vehicle.

Seat.Occupant:
  type: reference
  description: A reference to the Occupant as a node path name (e.g., Vehicle.Occupant.Person1).
    
```

Reply:
I like the idea, but it is something that needs to be discussed. I think it is somewhat aligned with some thoughts presented by Alan Freedman on making instance handling more flexible. Like treating "Row, Pos and Seat-Function (Driver-side, middle, Passenger-side) as labels which you depending on implementation can use to identify an instance. The possibility to support references is also something that has been discussed in the past, for example to support links from truck to trailer.

For references I think it might be relevant to specify the type of the reference, for example to specify that the reference is to a `Vehicle.Cabin.Seat` but how actual content is represented might depend on implementation. Some may use dot-notated path, others a URL or some other metadata that the client understand.